addons/test_mail/tests/test_mail_composer.py

```diff
@@ -423,6 +434,32 @@ def test_mail_composer_rights_portal(self):
             self.assertEqual(self.test_record.message_ids[0].body, '<p>Body text 2</p>')
             self.assertEqual(self.test_record.message_ids[0].author_id, portal_user.partner_id)
 
+    @users('user_rendering_restricted')
+    def test_mail_composer_rights_template(self):
+        template_1 = self.template.copy({
+            'report_name': 'TestReport for {{ object.name }}.html',  # test cursor forces html
+            'report_template': self.test_report.id,
+        })
+        attachment_data = self._generate_attachments_data(2)
+        template_1.write({
+            'attachment_ids': [(0, 0, dict(a, res_model="mail.template", res_id=template_1.id)) for a in attachment_data]
+        })
+
+        template_1_attachments = template_1.attachment_ids
+        self.assertEqual(len(template_1_attachments), 2)
+
+        composer = self.env['mail.compose.message'].with_context(
+            self._get_web_context(self.test_record)
+        ).create({
+            'subject': 'Template Subject',
+            'body': '<p>Template Body</p>',
+            'template_id': template_1.id,
+            'attachment_ids': template_1_attachments.ids,
+            'partner_ids': [self.partner_employee_2.id],
+        })
+        composer._action_send_mail()
```

Comment:
Could maybe check the result to check that we did not have ACL issue, but also that attachments are effectively correctly managed ^^